### PR TITLE
Remove `IRHLSLExportDecoration` and `IRKeepAliveDecoration` for non-CUDA/Torch targets

### LINKED ISSUE #4289: Slangc crashes when using  `[AutoPyBindCUDA]` and `[shader("compute")]` in the same file
Using slangtorch's loadModule on a slang file that defines both an `[AutoPyBindCUDA]` entry point and and a `[shader("compute")]`  entry point (different functions) causes `slangc` to crash ("*Compilation failed with error 3221225477*" in Python traceback, no stderr from slangc).

Example:

```csharp
[AutoPyBindCUDA]
[CUDAKernel]
void square(TensorView<float> input, TensorView<float> output)
{}

[shader("compute")]
[numthreads(1,1,1)]
void computeMain(uint3 threadId: SV_DispatchThreadID)
{}
```

### PR DESCRIPTION
**Pull request by saipraveenb25**
Fixes #4289.

### CODE REVIEW

File: tests/autodiff/hlsl-torch-cross-compile.slang [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    v[0] = func1(v[0]);
+    v[1] = func1(v[1]);
+}
+
+// Shouldn't see torchMain (or its transformations) anywhere in the HLSL output
+
+// HLSL: {{.*}}func1{{.*}}
+// HLSL-NOT: {{.*}}torchMain{{.*}}
+// HLSL: {{.*}}computeMain{{.*}}
```
Comment (jkwak-work) [inline tests/autodiff/hlsl-torch-cross-compile.slang:29]: I don't think you need to have `{{.*}}` at the beginning and the end of the line.
And you may want to add more "HLSL-NOT" case above and beyond like this:
```
// Shouldn't see torchMain (or its transformations) anywhere in the HLSL output
// HLSL-NOT:torchMain
// HLSL:func1
// HLSL-NOT:torchMain
// HLSL:computeMain
// HLSL-NOT:torchMain
```

**Review by jkwak-work — APPROVED** [review]
Looks good to me.

Is there an issue number related to this PR?

File: tests/autodiff/hlsl-torch-cross-compile.slang (hunk 1 above)
Comment (saipraveenb25) [reply tests/autodiff/hlsl-torch-cross-compile.slang:29]: Oh yeah.. this would be cleaner.

**Comment by saipraveenb25** [comment]
Yes, I've added the issue number now.

**Review by csyonghe — APPROVED** [review]

**Pull request merged**